src/crimson/os/seastore/extent_placement_manager.h

```diff
@@ -545,6 +545,12 @@ class ExtentPlacementManager {
     return background_process.run_until_halt();
   }
 
+  bool get_checksum_needed(paddr_t addr) {
+    // checksum offloading only for blocks physically stored in the device
+    assert(addr.is_absolute());
```

Comment:
Follow up https://github.com/ceph/ceph/pull/57782#discussion_r1673498868

Look deeper, probably this fails UTs is because there are fake paddrs.

If that's true, I think this can be adjusted to `assert(addr.is_absolute() || addr.is_fake());`. Probably we can even adjust `is_absolute() -> True` for the fake paddrs, which conceptually seems reasonable.

How about try this way and remove all the additional `is_absolute()` guards? 

This should be helpful to understand how SeaStore is really working related to this feature.